### PR TITLE
fix(watcher): reload completed config writes

### PR DESCRIPTION
**Pull request by xz-dev**
## Problem

The config watcher registers `config.yaml` itself with fsnotify. Atomic-save tools replace that inode with a rename, so later replacements can stop producing reloads. The existing 150 ms debounce also reloads Linux in-place writes before the writer closes, which can expose an incomplete YAML prefix.

## Compatibility goals

This is a compatibility fix for both existing config-update styles, not a new save protocol:

1. **In-place writes:** a writer opens and updates the existing `config.yaml` inode. This remains supported; on Linux, reload waits for `IN_CLOSE_WRITE` instead of guessing completion from a 150 ms timer.
2. **Atomic replacements:** a writer completes a temporary file and renames it over `config.yaml`. Watching the stable parent directory and accepting `IN_MOVED_TO` keeps every replacement observable after the original inode disappears.

Both paths enter the same serialized snapshot/validation/hash reload flow. Writers do not need to adopt a new API or file format.

## Established design precedent

This does not invent a new filesystem-watching method:

- [fsnotify's official documentation](https://github.com/fsnotify/fsnotify#watching-files) recommends watching the parent directory and filtering `Event.Name`, specifically because editors commonly replace files atomically and invalidate an inode-level watch.
- Go's Linux syscall layer already exposes the standard inotify operations and event masks, including `InotifyAddWatch`, `IN_CLOSE_WRITE`, and `IN_MOVED_TO`. This implementation uses the Go project's maintained [`golang.org/x/sys/unix`](https://pkg.go.dev/golang.org/x/sys/unix) wrapper for those same kernel primitives rather than defining a custom protocol.
- Released fsnotify versions do not expose close-write as a portable public operation, so the patch keeps fsnotify as the cross-platform watcher and adds only a small Linux completion adapter using the already-present Go-maintained package.

## Changes

- watch the config parent directory and filter the exact normalized target path, while avoiding duplicate registration when the auth directory is shared;
- on Linux, use `IN_CLOSE_WRITE` and `IN_MOVED_TO` as completion signals through a small `golang.org/x/sys/unix` adapter;
- fall back to the existing fsnotify quiet-window behavior on non-Linux platforms and whenever native Linux completion is unavailable or loses events;
- serialize config reloads and parse/hash one byte snapshot so the applied config and deduplication hash represent the same generation;
- preserve load-time plaintext management-secret migration without overwriting a newer atomic replacement, and restore original bytes after recoverable partial-write or truncate failures;
- wait for admitted config reload and config-persistence work during `Stop`;
- add real-filesystem regression tests for repeated atomic replacement, in-place close gating, native/fallback admission, overflow fallback, generation consistency, migration failures, and shutdown fencing.

## Compatibility

- missing-config `Start` failure remains unchanged;
- auth watching, explicit `ReloadConfigIfChanged`, hash deduplication, `LoadConfig`-equivalent validation, and management-secret migration remain supported;
- non-Linux platforms keep the existing 150 ms debounce fallback;
- no public API or dependency changes.

## Not included

- config transaction or revision/CAS support;
- atomic or durable config persistence redesign;
- Docker mount or remote-store changes;
- plugin business logic;
- changes under `internal/translator`.

## Verification

```text
go test ./internal/config ./internal/watcher/... -count=1
go test -race ./internal/config ./internal/watcher -count=1
go test ./internal/config ./internal/watcher -run 'Test(ParseConfigBytesAndPersistMigrations|ConfigCompletion|ConfigWatcher|StartCaptures|StartFallsBack|ReloadConfigIfChanged|StopWaits)' -count=100
GOOS=windows GOARCH=amd64 CGO_ENABLED=0 go test -c ./internal/watcher
GOOS=darwin GOARCH=amd64 CGO_ENABLED=0 go test -c ./internal/watcher
GOOS=freebsd GOARCH=amd64 CGO_ENABLED=0 go test -c ./internal/watcher
go test ./cmd/server -run '^$'
go build -o /tmp/cli-proxy-api ./cmd/server
go vet ./internal/config ./internal/watcher/...
git diff --check origin/dev..HEAD
git verify-commit HEAD
```

Independent final review approved the current diff.

`go test ./... -count=1` still reports three Claude fingerprint tests expecting Linux while the host fingerprint resolves to macOS. The same three failures reproduce on a clean `origin/dev` worktree; this patch adds no full-suite regression.

### CODE REVIEW

File: internal/watcher/config_completion_other.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+//go:build !linux
+
+package watcher
+
+import "context"
+
+type configCompletionWatcher struct{}
+
+func newConfigCompletionWatcher(string) (*configCompletionWatcher, error) {
+	return nil, nil
```
[the remark this answers is not in the document]
Comment (xz-dev) [reply internal/watcher/config_completion_other.go]: Fixed in b2b2d61d. The non-Linux constructor now returns a literal nil `configCompletion` interface, so `NewWatcher` keeps native completion inactive and fsnotify debounce fallback remains enabled. Added a `!linux` regression asserting the returned interface is nil; Windows, Darwin, and FreeBSD watcher test binaries compile successfully.

File: internal/watcher/events.go [hunk 2]
```diff
@@ -27,18 +28,61 @@ func matchProvider(provider string, targets []string) (string, bool) {
...
+	if w.startTestHook != nil {
+		w.startTestHook()
+	}
+	if _, errStatConfig := os.Stat(w.configPath); errStatConfig != nil {
+		closeCompletionOnError()
+		log.Errorf("failed to access config file %s: %v", w.configPath, errStatConfig)
+		return errStatConfig
+	}
+	configDir := filepath.Dir(w.configPath)
+	if errAddConfig := w.watcher.Add(configDir); errAddConfig != nil {
```
[the remark this answers is not in the document]
Comment (xz-dev) [reply internal/watcher/events.go]: Fixed in b2b2d61d. Watcher setup now resolves both the logical config path and its symlink target, watches each distinct parent directory, and filters exact paths. Linux completion similarly tracks deduplicated `(directory, basename)` targets and matches exact `(watch descriptor, basename)`. Real-filesystem regressions cover an in-place target write gated on close and repeated atomic target replacements.
